sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -265,6 +265,10 @@ private[hive] class HiveClientImpl(
       clientLoader.cachedHive.asInstanceOf[Hive]
     } else {
       val c = getHive(conf)
+      val metaConfPrefix = SQLConf.get.getConf(HiveUtils.HIVE_METASTORE_METACONF_PREFIX)
+      conf.getAllProperties.asScala
+        .filterKeys(_.startsWith(metaConfPrefix))
+        .foreach{ case(key, value) => c.setMetaConf(key.substring(metaConfPrefix.length), value)}
```

Comment:
Since we don't have any example here, could you elaborate more about the specific hive configurations you want to set? Please write it in the PR description, @kevincmchen .

Reply:
i have add the description, pls check

Reply:
in our case,  we pass tenant info to an external multi-tenant metasotre by utilizing the meta conf, such as tenant name, auth info etc.